### ISSUE TITLE
Launching PS1 gives RetroArch menu?

### ISSUE THREAD

**Issue by ultimateguy** [rommapp/romm#1209]
How does one launch a playstation 1 game? I just get the RetroArch menu. Figured I had to launch the game from RetroArch but my game folder does not seem to be mapped anywhere. Other platforms seem to work fine.

![image](https://github.com/user-attachments/assets/d3020b5e-a151-406e-88aa-f30e2e572a37)

**Comment by gantoine** [comment]
You need to make sure you're using a valid PS1 game and loading the right BIOS file

**Comment by lgwapnitsky** [comment]
I have my structure set up properly and have the BIOS files, but still get the same thing

**Comment by psychonic** [comment]
> I have my structure set up properly and have the BIOS files, but still get the same thing

I have the same issue. Did you ever figure it out?

**Comment by Spinnich** [comment]
> > I have my structure set up properly and have the BIOS files, but still get the same thing
> 
> I have the same issue. Did you ever figure it out?

Do you select the correct BIOS file when launching the game?

**Comment by psychonic** [comment]
Yes, I only have one to choose from in the menu. It's named `scph5501.bin`, and I had copied it from a working local Retroarch instance.

**Comment by Spinnich** [comment]
> Yes, I only have one to choose from in the menu. It's named `scph5501.bin`, and I had copied it from a working local Retroarch instance.

Sounds correct, could you provide as much detail as possible on the PS1 ROM you’re trying to launch?

**Comment by psychonic** [comment]
The issue happens with any PSX game I try (and only with PSX games).

One example: `Twisted Metal III (USA) (Rev 1)`, with trying to launch either with the m3u or the cue. (I realize that the m3u is redundant for single disc games, but I have it for all of mine).

SHA-1: `55c7c7739ffa4ca84f7b2018e73269a8e6f536b2`

With the pcsx_rearmed core, it launches to the Retroarch main menu.

With the mednafen_psx_hw core, it shows the error, `Failed to start game` on the emulator.js loading screen.

File list below. 

### File list
```
Twisted Metal III (USA) (Rev 1) (Track 01).bin
Twisted Metal III (USA) (Rev 1) (Track 02).bin
Twisted Metal III (USA) (Rev 1) (Track 03).bin
Twisted Metal III (USA) (Rev 1) (Track 04).bin
Twisted Metal III (USA) (Rev 1) (Track 05).bin
Twisted Metal III (USA) (Rev 1) (Track 06).bin
Twisted Metal III (USA) (Rev 1) (Track 07).bin
Twisted Metal III (USA) (Rev 1) (Track 08).bin
Twisted Metal III (USA) (Rev 1) (Track 09).bin
Twisted Metal III (USA) (Rev 1) (Track 10).bin
Twisted Metal III (USA) (Rev 1) (Track 11).bin
Twisted Metal III (USA) (Rev 1) (Track 12).bin
Twisted Metal III (USA) (Rev 1) (Track 13).bin
Twisted Metal III (USA) (Rev 1) (Track 14).bin
Twisted Metal III (USA) (Rev 1) (Track 15).bin
Twisted Metal III (USA) (Rev 1) (Track 16).bin
Twisted Metal III (USA) (Rev 1) (Track 17).bin
Twisted Metal III (USA) (Rev 1) (Track 18).bin
Twisted Metal III (USA) (Rev 1) (Track 19).bin
Twisted Metal III (USA) (Rev 1) (Track 20).bin
Twisted Metal III (USA) (Rev 1) (Track 21).bin
Twisted Metal III (USA) (Rev 1) (Track 22).bin
Twisted Metal III (USA) (Rev 1) (Track 23).bin
Twisted Metal III (USA) (Rev 1) (Track 24).bin
Twisted Metal III (USA) (Rev 1) (Track 25).bin
Twisted Metal III (USA) (Rev 1) (Track 26).bin
Twisted Metal III (USA) (Rev 1) (Track 27).bin
Twisted Metal III (USA) (Rev 1) (Track 28).bin
Twisted Metal III (USA) (Rev 1) (Track 29).bin
Twisted Metal III (USA) (Rev 1) (Track 30).bin
Twisted Metal III (USA) (Rev 1) (Track 31).bin
Twisted Metal III (USA) (Rev 1) (Track 32).bin
Twisted Metal III (USA) (Rev 1) (Track 33).bin
Twisted Metal III (USA) (Rev 1) (Track 34).bin
Twisted Metal III (USA) (Rev 1) (Track 35).bin
Twisted Metal III (USA) (Rev 1) (Track 36).bin
Twisted Metal III (USA) (Rev 1) (Track 37).bin
Twisted Metal III (USA) (Rev 1).cue
Twisted Metal III (USA) (Rev 1).m3u
```

**Comment by Spinnich** [comment]
> The issue happens with any PSX game I try (and only with PSX games).
> 
> One example: `Twisted Metal III (USA) (Rev 1)`, with trying to launch either with the m3u or the cue. (I realize that the m3u is redundant for single disc games, but I have it for all of mine).
> 
> SHA-1: `55c7c7739ffa4ca84f7b2018e73269a8e6f536b2`
> 
> With the pcsx_rearmed core, it launches to the Retroarch main menu.
> 
> With the mednafen_psx_hw core, it shows the error, `Failed to start game` on the emulator.js loading screen.
> 
> File list below.
> 
> ### File list
> ```
> Twisted Metal III (USA) (Rev 1) (Track 01).bin
> Twisted Metal III (USA) (Rev 1) (Track 02).bin
> Twisted Metal III (USA) (Rev 1) (Track 03).bin
> Twisted Metal III (USA) (Rev 1) (Track 04).bin
> Twisted Metal III (USA) (Rev 1) (Track 05).bin
> Twisted Metal III (USA) (Rev 1) (Track 06).bin
> Twisted Metal III (USA) (Rev 1) (Track 07).bin
> Twisted Metal III (USA) (Rev 1) (Track 08).bin
> Twisted Metal III (USA) (Rev 1) (Track 09).bin
> Twisted Metal III (USA) (Rev 1) (Track 10).bin
> Twisted Metal III (USA) (Rev 1) (Track 11).bin
> Twisted Metal III (USA) (Rev 1) (Track 12).bin
> Twisted Metal III (USA) (Rev 1) (Track 13).bin
> Twisted Metal III (USA) (Rev 1) (Track 14).bin
> Twisted Metal III (USA) (Rev 1) (Track 15).bin
> Twisted Metal III (USA) (Rev 1) (Track 16).bin
> Twisted Metal III (USA) (Rev 1) (Track 17).bin
> Twisted Metal III (USA) (Rev 1) (Track 18).bin
> Twisted Metal III (USA) (Rev 1) (Track 19).bin
> Twisted Metal III (USA) (Rev 1) (Track 20).bin
> Twisted Metal III (USA) (Rev 1) (Track 21).bin
> Twisted Metal III (USA) (Rev 1) (Track 22).bin
> Twisted Metal III (USA) (Rev 1) (Track 23).bin
> Twisted Metal III (USA) (Rev 1) (Track 24).bin
> Twisted Metal III (USA) (Rev 1) (Track 25).bin
> Twisted Metal III (USA) (Rev 1) (Track 26).bin
> Twisted Metal III (USA) (Rev 1) (Track 27).bin
> Twisted Metal III (USA) (Rev 1) (Track 28).bin
> Twisted Metal III (USA) (Rev 1) (Track 29).bin
> Twisted Metal III (USA) (Rev 1) (Track 30).bin
> Twisted Metal III (USA) (Rev 1) (Track 31).bin
> Twisted Metal III (USA) (Rev 1) (Track 32).bin
> Twisted Metal III (USA) (Rev 1) (Track 33).bin
> Twisted Metal III (USA) (Rev 1) (Track 34).bin
> Twisted Metal III (USA) (Rev 1) (Track 35).bin
> Twisted Metal III (USA) (Rev 1) (Track 36).bin
> Twisted Metal III (USA) (Rev 1) (Track 37).bin
> Twisted Metal III (USA) (Rev 1).cue
> Twisted Metal III (USA) (Rev 1).m3u
> ```

Hrm, I never use multi-file ROMs like this within EmulatorJS so I'm not sure on the nuance for making them work but they _should_ according to some documentation. That said, as a sanity check, I would convert one of your games to .chd and see if that behaves better, I suspect it will and at least narrows down your issue until your figure out how to make EmulatorJS play nicely with ROMs in that format.

**Comment by psychonic** [comment]
Ah, that makes sense. Is it possible that RomM is only making the selected file available to EmulatorJS, rather than the whole folder contents?

Either way, I'll look into converting to CHD and see if that work. Will post back here afterward either way to confirm. Thanks for the help!

**Comment by Spinnich** [comment]
> Ah, that makes sense. Is it possible that RomM is only making the selected file available to EmulatorJS, rather than the whole folder contents?
> 
> Either way, I'll look into converting to CHD and see if that work. Will post back here afterward either way to confirm. Thanks for the help!

Hrm, it's tough for me to say personally as I tend to favor formats that are emulator friendly (i.e. .chd) over formats that are archive pure and I rarely use the baked in EmulatorJS functionality to boot. I wouldn't be surprised that if you poke around on the RomM Discord you could find people more familiar with this combo however to help you out!

**Comment by psychonic** [comment]
I can confirm that it works fine with CHD files. The `Loading game data...` screen took noticeably longer as well (although still quick), which, while not a scientific test, may give some weight to the theory that it only ever loads the selected file rather than all of the files for a game.

**Comment by psychonic** [comment]
I dug into the code a bit and figured out what's going on. It actually works fine with bin/cue files as long as you don't specify a file on the launch screen. (I didn't realize it was optional). It then does grab all of the files in the folder.
